server/client/ruby/candlepin.rb

```diff
@@ -387,6 +393,26 @@ def set_scheduler_status(opts = {})
       def delete_job(opts = {})
         delete_by_id("/jobs", :job_id, opts)
       end
+
+      def wait_for_job(opts = {})
+        defaults = {
+          :job_id => nil,
+        }
+        opts = verify_and_merge(opts, defaults)
+
+        res = get_job(:job_id => opts[:job_id])
+        loop do
+          job = res.ok_content
+          case job[:state]
+          when "CREATED", "PENDING", "WAITING", "RUNNING"
+            sleep 1
+            res = get_job(:job_id => opts[:job_id])
+          when "FINISHED", "CANCELED", "FAILED"
```

Comment:
> @awood not a big deal since you have covered all states, by why not "else" here? if you had not captured all the states, this would have ended up in an infinite loop...

Added.  I didn't force push so this comment isn't marked as out-dated.